sw-emulator/lib/periph/src/soc_reg.rs

```diff
@@ -1142,7 +1106,11 @@ impl Bus for SocRegistersImpl {
     /// Called by Bus::poll() to indicate that time has passed
     fn poll(&mut self) {
         if self.timer.fired(&mut self.op_fw_write_complete_action) {
-            self.upload_firmware().unwrap();
+            (self.ready_for_fw_cb)(&mut self.mailbox, &self.firmware);
```

Comment:
self.firmware? Should that come from the app?

Reply:
it is passed as an argument by the app.

Reply:
Now it is coming from the app environment - removed the argument passing to the model